pkg/visor/logserver/api.go

```diff
@@ -32,114 +27,200 @@ type API struct {
 	startedAt time.Time
 }
 
-// New creates a new api.
+// New creates a new API.
 func New(log *logging.Logger, tpLogPath, localPath, customPath string, whitelistedPKs []cipher.PubKey, printLog bool) *API {
 	api := &API{
 		logger:    log,
 		startedAt: time.Now(),
 	}
-
-	r := chi.NewRouter()
-
-	r.Use(middleware.RequestID)
-	r.Use(middleware.RealIP)
+	// disable gin's debug logging on startup
+	gin.SetMode(gin.ReleaseMode)
+	// Gin router without default logging
+	r := gin.New()
+	// use Gin's recovery logging middleware to recover from panic
+	r.Use(gin.Recovery())
 	if printLog {
-		r.Use(middleware.Logger)
-		r.Use(middleware.Recoverer)
+		// use custom logging middleware
+		r.Use(loggingMiddleware())
 	}
-	r.Use(httputil.SetLoggerMiddleware(log))
-
-	r.Get("/health", api.health)
 
-	fsTP := http.FileServer(http.Dir(tpLogPath))
-	r.Handle("/*", http.StripPrefix("/", fsTP))
-
-	fsLocal := http.FileServer(http.Dir(localPath))
-	r.Handle("/"+skyenv.NodeInfo, http.StripPrefix("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		whitelistFSHandler(w, r, localPath, whitelistedPKs)
-	})))
+	// whitelist-based authentication for survey collection if there are keys whitelisted for that
+	// no survey-whitelisted keys means the file is publicly accessible
+	authRoute := r.Group("/")
+	if len(whitelistedPKs) > 0 {
+		authRoute.Use(whitelistAuth(whitelistedPKs))
+	}
+	// note that the survey only exists / is generated if the reward address is set
+	authRoute.StaticFile("/node-info.json", filepath.Join(localPath, "node-info.json"))
 
-	r.Handle("/transport_logs/*", http.StripPrefix("/", fsLocal))
+	r.GET("/health", func(c *gin.Context) {
+		api.health(c)
+	})
 
-	fsCustom := http.FileServer(http.Dir(customPath))
-	r.Handle("/*", http.StripPrefix("/", fsCustom))
+	// serve transport log files ; then any files in the custom path
+	r.GET("/:file", func(c *gin.Context) {
+		// files with .csv extension are likely transport log files
+		if filepath.Ext(c.Param("file")) == ".csv" {
```

Comment:
Having custom logic based on the file extension sound like a bad idea as it is unexpected.

Reply:
This was already sortof a mess because we have multiple dirs serving to "/"

We only are serving .csv files for transport logs, so it makes sense to check there first and only in the instance that the extension matches

if it is not a .csv, then we serve the custom dmsghttp path.

basically it just prevents the transport logs dir from being used to serve anything else besides transport logs